src/table.rs

```diff
@@ -202,6 +202,19 @@ impl Table {
     pub fn decor(&self, key: &str) -> Option<&Decor> {
         self.items.get(key).map(|kv| &kv.key.decor)
     }
+
+    /// Sets the position of the `Table` within the `Document`.
+    ///
+    /// This will never create invalid toml but, formatting may not be
+    /// preserved (see `tests::test_edit::test_multiple_max_usize_positions`).
+    pub fn set_position(&mut self, position: usize) {
+        self.position = Some(position);
+    }
+
+    /// The position of the `Table` within the `Document`.
```

Comment:
It's not clear from a user's perspective, what `None` means.
Though I'm not happy with leaking implementation detail to the public API
```suggestion
    /// The position of the `Table` within the `Document`.
    ///
    /// Returns `None` if the `Table` was created manually (i.e. not via parsing)
    /// in which case its position is set automatically.
```